File: Java/HLC02_EjemplosCodigo01_09/07_GuardarEstadoNo/app/src/main/java/com/pms/a07_guardarestadono/MainActivity.java
package com.pms.a07_guardarestadono;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    int contador=0;
    TextView texto;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        //referenciamos la vista TextView del layout
        texto=(TextView) findViewById(R.id.text);
    }

    /**
     * Comportamiento del botón suma
     * @param view: botón suma
     */

    public void onClickSuma(View view){
        //incrementa 1 el contador
        contador++;
        //asigna el valor del contador al TextView
        texto.setText(String.valueOf(contador));
    }

}

File: Kotlin/HLC02_EjemplosCodigo01_09/07_GuardarEstadoNo/app/src/main/java/com/pms/a07_guardarestadono/MainActivity.kt
package com.pms.a07_guardarestadono

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.View
import android.widget.TextView

class MainActivity : AppCompatActivity() {
    var contador = 0
    var texto: TextView? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        //referenciamos la vista TextView del layout
        texto = findViewById<View>(R.id.text) as TextView
    }

    /**
     * Comportamiento del botón suma
     * @param view: botón suma
     */
    fun onClickSuma(view: View?) {
        //incrementa 1 el contador
        contador++
        //asigna el valor del contador al TextView
        texto!!.text = contador.toString()
    }
}